### PR TITLE
Fixes #21646, #21647 Rework architecture, api changes, redux store

### PR DESCRIPTION
**Pull request by danseethaler**
This PR overhauls the directory structure, mock api, and adds redux structure. It depends on https://github.com/theforeman/foreman/pull/5014. I also had to bump the `patternfly-react` package in theforeman. You can see the version bump in katello `0.8.1` to `0.11.0`.

Please review file names, redux structure, and overall architecture. Any other comments are welcome. This is still a WIP (needs real API integration, actions on list items, etc.)

Still need to add some tests after we get the components right.

### CODE REVIEW

**Comment by danseethaler** [comment]
Let me know if anyone has any ideas on the HoundCI errors.  The `import/no-unresolved` is explicitly turned off in the `.eslintrc` file since Hound throws invalid errors.

**Comment by waldenraines** [comment]
I know this isn't related to this PR but would you mind [changing this](https://github.com/Katello/katello/blob/master/webpack/services/index.js#L8) from `rpm` to `yum` so that the icons display for yum repositories?  

Also, @rohoover, what icon should we use for file repositories?

File: package.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
     "bootstrap-select": "^1.12.2",
     "classnames": "^2.2.5",
     "patternfly": "^3.29.5",
-    "patternfly-react": "^0.8.1",
+    "patternfly-react": "^0.11.0",
```
Comment (waldenraines) [inline package.json]: May as well make this `^0.13.0` since they've released that.

File: webpack/index.js [hunk 2]
```diff
@@ -6,7 +6,7 @@
 import componentRegistry from 'foremanReact/components/componentRegistry';
 import { mount } from 'foremanReact/common/MountingService';
 import ExperimentalUi from './containers/Application/index';
-import './reducers';
+import './redux/reducers';
```
Comment (waldenraines) [inline webpack/index.js]: 👍  to changing this to be consistent with foreman.

File: webpack/scenes/RedHatRepositories/components/EnabledRepositoriesList.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import EnabledRepository from './EnabledRepository';
+
+const EnabledRepositories = ({ repositorySets }) =>
+  repositorySets.map(({ type, repositories }) =>
+    repositories.map(repo => <EnabledRepository key={repo.id} type={type} {...repo} />));
+
+EnabledRepositories.propTypes = {
+  repositorySets: PropTypes.arrayOf(PropTypes.shape({
+    repositories: PropTypes.arrayOf(PropTypes.object).isRequired,
+    type: PropTypes.string.isRequired, // 'kickstart',
```
Comment (waldenraines) [inline webpack/scenes/RedHatRepositories/components/EnabledRepositoriesList.js]: I wonder about these comments.  I can see both their usefulness and their superfluity 😄

Comment (danseethaler) [reply webpack/scenes/RedHatRepositories/components/EnabledRepositoriesList.js]: Yeah I think they could be removed now or in the near future. I left them in there when I copied over the api data to give me some bearings :)

**Comment by danseethaler** [comment]
@waldenraines one idea on the api mocking would be to have the [get](https://github.com/Katello/katello/pull/7092/files#diff-e219692c9292fe6766f9500daa20ec15R6) method accept the api call name (i.e. `PRODUCT_REPOSITORY_SETS`) and any necessary parameters (org id, product id, etc) and then it will either construct an api call using `fetch` or returned the mock data associated with that api name.

This would enable us to work with the mocked data while the api is being developed and in the testing environment. The result of this would be having a central place where all api calls are defined. I don't know if that would be a nice feature or technical debt.

Thoughts?

File: webpack/index.js (hunk 2 above)
Comment (Rohoover) [reply webpack/index.js]: Hey -- So are we talking an icon that represents repos, or one that represents files in repos?

**Comment by Rohoover** [comment]
fa-code

**Pull request merged**